pkg/obfuscator/report_types.go

```diff
@@ -0,0 +1,31 @@
+package obfuscator
+
+type ReplacementReport struct {
+	Replacements []Replacement
+}
+
+type Replacement struct {
+	Canonical    string       `yaml:"canonical,omitempty"`
+	ReplacedWith string       `yaml:"replacedWith,omitempty"`
+	Occurrences  []Occurrence `yaml:"occurrences,omitempty"`
+}
+
+type Occurrence struct {
+	Original string `yaml:"original,omitempty"`
+	Count    uint   `yaml:"count,omitempty"`
+}
+
+func (r *Replacement) Increment(original string, count uint) Replacement {
```

Comment:
you should definitely add some tests for this method

Reply:
and why do you return the Replacement?

Reply:
I mean, you already have a pointer to the replacement, you can just increment the counter directly without returning it.

Reply:
:+1: on tests

I'll review how Replacement is used

Reply:
also below in the reporting is nothing else like a good old std::multimap. I'm sure there must be a good library for us to reuse?